src/firefly/js/charts/dataTypes/XYColsCDT.js

```diff
@@ -194,7 +194,9 @@ export function setZoom(chartId, chartDataElementId, selection=undefined) {
             if (decimatedUnzoomed || isUndefined(decimatedUnzoomed)) {
                 noFetch = false;
             }
-            dispatchChartOptionsUpdate({chartId, chartDataElementId, updates: {zoom: xyPlotParams.selection, selection: undefined}, noFetch});
+            dispatchChartOptionsUpdate({chartId, chartDataElementId, updates: {zoom: xyPlotParams.boundaries, selection: undefined}, noFetch});
```

Comment:
Your approach "if the zoom is in the old options, copy it to the new options" is correct. Just do it in XYPlotOptions and only use "zoom" field. Currently, every time we press "Apply" button in options dialog, zoom is gone. We want to preserve it, unless X or Y columns changed.

We use "zoom" field to track zoom and "boundaries" field to track data or user boundaries. "boundaries" is a calculated field and should never be set directly. 

Because you clear boundaries, not zoom, the unzoom icon does not disappear when the chart is unzoomed.

When I said that using boundaries messes up setting min and max values, I was wrong. I noticed that zooming up and then setting min/max value does not work. But the real cause of this behavior is a bug in lines 376-377 of XYPlot. If both zoom and data boundaries are present, min/max of the two should be taken as an axis min/max. I would appreciate if you fix it too. (Or I can fix it later in your branch.)